include/rocksdb/utilities/table_properties_collectors.h

```diff
@@ -130,4 +132,81 @@ class CompactForTieringCollectorFactory
 
 std::shared_ptr<CompactForTieringCollectorFactory>
 NewCompactForTieringCollectorFactory(double compaction_trigger_ratio);
+
+// Information for the age of a collection of data. The unit for all the age
+// stats are in seconds. Data's age is defined as the time passed since the data
+// was initially written into the DB. Check `DataCollectionIsEmpty` and
```

Comment:
Does this mean the structure becomes less accurate over time, since data with age 1000 seconds needs to be 2000 seconds in another 1000 seconds? Any reason not to use unix times?

Reply:
My worry is that the struct makes it a little too easy to "do the wrong thing" such as saving the information for minutes or hours or days and forgetting that it is out of date. I would be satisfied if the struct contains a field for the unix time that the ages are relative to. Or maybe it's better for the struct to store the data as unix times and expose the ages through public functions, perhaps relative to a specified "current time."

I see this as somewhat analogous to my argument for using UTC time in `daily_offpeak_time_utc`. The simplest, most "convenient" API is usually best but not if it invites bugs. Prompting the user to think about the potential bug-prone piece (what is the reference point for these times?) is the best approach IMHO.

> I figured this is a more intuitive definition of data's age, if we directly use unix time, I'm not sure the definition of averaging this unix time age is as meaningful.

(Assuming exclusion of "infinitely old"...) Yeah, it may not be obvious to general users that average/min/max absolute times is equivalent to average/min/max relative time (even though these are robust under linear transformations and then the reverse transformation)

Reply:
Thanks for the very detailed context! Yeah, unix time is really standard practices. The seemingly convenient `age` API I was trying to provide is really unnecessary and dangerous design as you pointed out. Users can easily do a (now() - write_time) query to get age themselves. I should have been a non-general user myself to understand that absolute time and relative time work the same for these statistics, thanks for calling that out!